lib/api/paging/time_paginator.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Api
+  module Paging
+    class TimePaginator
```

Comment:
It's more a question of the data available for building the previous page when doing the paging.  The default searches right now with reverse chronological time are all "before" searches with descending times so that you get the most recent stuff on top.  So for example I start with "now" and get everything before "now" up to a certain number of items.  Then the next page would be before whatever the earliest result of this page, which would be the last item etc.  You can't start with "after" to get to now and go ascending.  

The ascending is for if you need to traverse back a page.  Again from the current page you are on you don't know how far "before" would have to be but you do know what the latest date on this page is so you can query for "after the last time here" up to a limit to get back to the previous page. The order that is returned is now ascending though but it actually is consistent with going back and forth in a timeline.

Reply:
Ok thank you for the explanation.